### PR TITLE
First steps for refactor: generators, vars, arrow funcs

### LINKED ISSUE #24: Get rid of generators, "var"

### PR DESCRIPTION
**Pull request by oleg-koval**
## Motivation

fixes #24 

## Changes

- move from generators to async/await in tests
- stop using `var`, but `const` & `let` instead
- use of arrow functions

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,25 +1,38 @@
...
+    return data;
+  }
+};
 
 /*
  * If logger is a bunyan logger instance, return it;
  * otherwise, create a new logger with some reasonable defaults.
  */
-function createOrUseLogger(logger) {
+const createOrUseLogger = logger => {
```
Comment (sjmcdowall) [inline index.js]: I sort of like the idea of having () around the args .. so = (logger) => {}  foo?  Many linters sort of demand this ..

File: index.js [hunk 2]
```diff
@@ -30,10 +43,10 @@ function createOrUseLogger(logger) {
  *                    that will be passed to bunyan.createLogger. If not
  *                    specified, a default logger will be used.
  */
-module.exports = function (loggerInstance) {
+module.exports = loggerInstance => {
```
Comment (sjmcdowall) [inline index.js]: See above

File: test/index.js [hunk 3]
```diff
@@ -36,119 +33,119 @@ describe('koaBunyanLogger', function () {
...
+  const request = () => {
     if (!server) {
       server = app.listen(0);
     }
 
     return supertest(server);
   };
 
-  var record = function (i) {
+  const record = (i) => {
```
Comment (sjmcdowall) [inline test/index.js]: Here's an example of how I like it :)

File: index.js (hunk 1 above)
Comment (oleg-koval) [reply index.js]: true that but we don't have any modern lining rules yet, so we can agree on such rule, will update the PR, the point is not even having parantheses or not, but have consistent codebase

0_dwiyyjzi5mu13l_m

Comment (sjmcdowall) [reply index.js]: LOL -- too true.  And I am looking at your LINT rules too :)

**Pull request merged**